### ISSUE TITLE
Enormous apk size increase (9MB -> 150MB)

### ISSUE THREAD

**Issue by ColorfulRhino** [Crazy-Marvin/MetadataRemover#105]
Something is wrong with the latest apk. This has also been talked about here https://github.com/Crazy-Marvin/MetadataRemover/issues/34#issuecomment-2231929176.

**Size comparison:**
v2.0.2: 9.29 MB
v2.0.3: 151 MB

More than 15x increase (!). Something's wrong, maybe a debug build was accidentally published or some recent commit added some unneccessarily large library?

**Comment by CrazyMarvin** [comment]
We recently added the ability to remove metadata from some documents and videos that need some bigger dependencies. 
Do you see any way to make the builds smaller, @Rashxz?

**Comment by Rashxz** [comment]
Yes, the increase is comming from the ffmpeg library, Ill see how to reduce the binary size.

**Comment by ColorfulRhino** [comment]
Thanks for clearing that up! I'm sure there is some way.
In my opinion (!) it's totally not worth it for one or two features to have a >1500% size increase.

**Comment by Rashxz** [comment]
I manage to build a new binary, new size per ABI is roughly 30 MB, universal APK is 40MB.

**Comment by ColorfulRhino** [comment]
> I manage to build a new binary, new size per ABI is roughly 30 MB, universal APK is 40MB.

That's an improvement 👍
Still quite an increase and too big for Izzy's repo, but well done regardless.

**Comment by TPS** [comment]
> Yes, the increase is comming from the ffmpeg library, Ill see how to reduce the binary size.

> > I manage to build a new binary, new size per ABI is roughly 30 MB, universal APK is 40MB.
> 
> That's an improvement 👍
> Still quite an increase and too big for Izzy's repo, but well done regardless.

Is it possible to get the same coverage by using using a _different_ library, e.g., [`exiftool`](https://github.com/exiftool/exiftool)?  On my `termux`, that's ¼ the installation size of `ffmpeg`, but has [a `perl` dependency](https://github.com/termux/termux-packages/blob/refs%2Fheads%2Fmaster/packages%2Fexiftool%2Fbuild.sh), so might come out even?

**Comment by Rashxz** [comment]
> > Yes, the increase is comming from the ffmpeg library, Ill see how to reduce the binary size.
> 
> > > I manage to build a new binary, new size per ABI is roughly 30 MB, universal APK is 40MB.
> > 
> > 
> > That's an improvement 👍
> > Still quite an increase and too big for Izzy's repo, but well done regardless.
> 
> Is it possible to get the same coverage by using using a _different_ library, e.g., [`exiftool`](https://github.com/exiftool/exiftool)? On my `termux`, that's ¼ the installation size of `ffmpeg`, but has [a `perl` dependency](https://github.com/termux/termux-packages/blob/refs%2Fheads%2Fmaster/packages%2Fexiftool%2Fbuild.sh), so might come out even?

The size its not comming only from ffmpeg, but also from the apache-poi library. It indeed could be a reeplacement as it supports a wide range of formats, but it wont help reducing the app size. 

Someone mentioned a already made port for android in this [issue](https://github.com/exiftool/exiftool/issues/129) and that one weights about 40MB, so our current solution is still even / maybe lighter,

**Comment by TPS** [comment]
@Rashxz The thing about `ffmpeg` (oh, did you see #117?) is that it's bit overkill for this project, since we don't need >90% of its capabilities here, ~~&`ffmpeg-kit` doesn't seem to allow for modular import of just the needed sections,~~ so that's just frustrating bloat.

**Comment by TPS** [comment]
@Rashxz Wait, maybe 1 _can_ decrease that bloat.…  Did you see https://github.com/arthenica/ffmpeg-kit/wiki/How-to-Decrease-Binary-Size?

**Comment by Rashxz** [comment]
> @Rashxz Wait, maybe 1 _can_ decrease that bloat.… Did you see https://github.com/arthenica/ffmpeg-kit/wiki/How-to-Decrease-Binary-Size?

![Captura de pantalla 2024-08-07 a la(s) 9 34 58 a  m](https://github.com/user-attachments/assets/76461d2d-d9c9-4418-9674-b477f1d9801e)

Yes, I build a new binary with everything stripped. As you can see in the Image, ffmpeg takes 6.9MB of the total size. The rest is coming from all the other dependencies, being the biggest ones apache poi (for libre office) and iText (for pdf).

I will take a look on the dependencies to see if theres room for improvement.

**Comment by TPS** [comment]
> Yes, I build a new binary with everything stripped. As you can see in the Image, ffmpeg takes 6.9MB of the total size. The rest is coming from all the other dependencies, being the biggest ones apache poi (for libre office) and iText (for pdf).

@Rashxz I'm certain there's cheaper avenues to take for these.  For pdf, may I suggest [`qpdf`](https://github.com/qpdf/qpdf) or [`cpdf`](https://github.com/coherentgraphics/cpdf-binaries#ccpythonnetjava-api) as alternatives?  I'll research for an alt for office docs, as well.

**Comment by TPS** [comment]
> > Yes, I build a new binary with everything stripped. As you can see in the Image, ffmpeg takes 6.9MB of the total size. The rest is coming from all the other dependencies, being the biggest ones apache poi (for libre office) and iText (for pdf).
> 
>I'll research for an alt for office docs, as well.

@Rashxz I wonder if a separate dedicated utility is actually needed for LO?  Per https://superuser.com/questions/1195338/what-metadata-does-a-libreoffice-file-contain-and-how-to-remove-it, it's matter of "just" deleting 1 file & 2 lines out of another out of the doc (as LO docs are actually zip archives of a certain structure).

**Comment by Rashxz** [comment]
> https://superuser.com/questions/1195338/what-metadata-does-a-libreoffice-file-contain-and-how-to-remove-it

Thanks

**Comment by Rashxz** [comment]
> > Yes, I build a new binary with everything stripped. As you can see in the Image, ffmpeg takes 6.9MB of the total size. The rest is coming from all the other dependencies, being the biggest ones apache poi (for libre office) and iText (for pdf).
> 
> @Rashxz I'm certain there's cheaper avenues to take for these. For pdf, may I suggest [`qpdf`](https://github.com/qpdf/qpdf) or [`cpdf`](https://github.com/coherentgraphics/cpdf-binaries#ccpythonnetjava-api) as alternatives? I'll research for an alt for office docs, as well.

Thanks

**Comment by Rashxz** [comment]
> > > Yes, I build a new binary with everything stripped. As you can see in the Image, ffmpeg takes 6.9MB of the total size. The rest is coming from all the other dependencies, being the biggest ones apache poi (for libre office) and iText (for pdf).
> > 
> > 
> > I'll research for an alt for office docs, as well.
> 
> @Rashxz I wonder if a separate dedicated utility is actually needed for LO? Per https://superuser.com/questions/1195338/what-metadata-does-a-libreoffice-file-contain-and-how-to-remove-it, it's matter of "just" deleting 1 file & 2 lines out of another out of the doc (as LO docs are actually zip archives of a certain structure).

Sorry, I did not specify it well in my previous comment, that library is not just for LO but also for Microsoft office docs.

**Comment by TPS** [comment]
@Rashxz How did https://github.com/Crazy-Marvin/MetadataRemover/commit/6928c9e100fd96e332f2e88ea0ee897d2fe0913f effect APK/installed size?

**Comment by Rashxz** [comment]
> @Rashxz How did [6928c9e](https://github.com/Crazy-Marvin/MetadataRemover/commit/6928c9e100fd96e332f2e88ea0ee897d2fe0913f) effect APK/installed size?

I think its a ~2MB increase

**Comment by TPS** [comment]
IIUC, https://github.com/Crazy-Marvin/MetadataRemover/issues/104#issuecomment-2270767530 notes all this is blocked from landing in F-droid, so no way to see these improvements for such users.

**Comment by CrazyMarvin** [comment]
This should be fixed.

**Comment by kryoqa** [comment]
Is video support no longer in the app ?
